package/MDAnalysis/analysis/base.py

```diff
@@ -281,26 +298,51 @@ def run(self, start=None, stop=None, step=None, verbose=None):
             stop frame of analysis
         step : int, optional
             number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory
         verbose : bool, optional
             Turn on verbosity
+
+
+    Example
+    -------
+       import MDAnalysis as mda
+       from MDAnalysisTests.datafiles import TPR, XTC
+       from MDAnalysis.analysis import rms
+
+       u = mda.Universe(TPR, XTC)
+
+       rmsd = rms.RMSD(u, select='name CA')
+       # A list of frames to run the analysis on can be passed
+       rmsd.run(frames=[0,2,3,6,9])
+       # A list of booleans the same length of the trajectory can also be used
+       rmsd.run(frames=[True, False, True, True, False, False, True, False,
+                        False, True])
+
+
+        .. versionchanged:: 2.1.0
+            Added ability to iterate through trajectory by passing a list of
+            frame indices
         """
         logger.info("Choosing frames to analyze")
         # if verbose unchanged, use class default
         verbose = getattr(self, '_verbose',
                           False) if verbose is None else verbose
 
-        self._setup_frames(self._trajectory, start, stop, step)
+        self._setup_frames(self._trajectory, start=start, stop=stop,
+                           step=step, frames=frames)
         logger.info("Starting preparation")
         self._prepare()
         for i, ts in enumerate(ProgressBar(
-                self._trajectory[self.start:self.stop:self.step],
+                self._sliced_trajectory,
                 verbose=verbose)):
             self._frame_index = i
             self._ts = ts
             self.frames[i] = ts.frame
             self.times[i] = ts.time
-            # logger.info("--> Doing frame {} of {}".format(i+1, self.n_frames))
             self._single_frame()
+        # rewind added for behavioral consistency, Issue #3416
+        self._trajectory.rewind()
```

Comment:
Thanks, could you just add a quick test to check that this is working as intended for now? (nothing fancy, just a quick check that the trajectory rewinds back to 0 will do)

Reply:
done